File: packages/sdk-python/examples/modify_command.py
import os
import polyhook

event = polyhook.read()

if event.tool == "bash" and os.environ.get("DRY_RUN"):
    cmd = (event.input or {}).get("command", "")
    polyhook.respond(polyhook.modify({"command": f'echo "would run: {cmd}"'}))
else:
    polyhook.respond(polyhook.approve())

File: packages/sdk-ts/examples/modify-command.ts
import { read, respond, modify, approve } from "@polyhook/sdk";

const event = await read();

// Prefix all bash commands with "echo would run: " in dry-run mode
if (event.tool === "bash" && process.env.DRY_RUN) {
  const cmd = (event.input?.command as string) ?? "";
  await respond(modify({ command: `echo "would run: ${cmd}"` }));
} else {
  await respond(approve());
}
